src/serlio/util/MArrayIteratorTraits.h

```diff
@@ -0,0 +1,110 @@
+/**
+ * Serlio - Esri CityEngine Plugin for Autodesk Maya
+ *
+ * See https://github.com/esri/serlio for build and usage instructions.
+ *
+ * Copyright (c) 2012-2019 Esri R&D Center Zurich
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "maya/MArrayIteratorTemplate.h"
+
+#include <iterator>
+
+namespace mu {
+
+namespace detail {
+
+// remove in favour of std::void_t once we switch to C++17
+template <typename... T>
+using void_t = void;
+
+template <typename Iterator, typename Default, typename = void_t<>>
+struct MArrayIteratorCategory {
+	typedef Default type;
```

Comment:
any reason not using 'using' everywhere?

Reply:
I usually use `typedef` just out of habit, however, for `void_t` I had to use `using` since it is templated.